### ISSUE TITLE
Test if process is defined in browser

### ISSUE THREAD

**Issue by pelikhan** [graphql/graphql-js#2676]
Hello all!

First thanks for this very well written library. It's a breeze to work with it.

I am hitting a snag when compiling graphql into a ES6 library. The node.js ``process`` object is queried in a mjs file which crashes in the browser. It would be great to add a check to make sure process is defined.

https://github.com/graphql/graphql-js/blob/2736f457b716b250c446b072c910f97e19943568/src/jsutils/instanceOf.js#L14

Thanks!

**Comment by IvanGoncharov** [comment]
@pelikhan Thanks for taking the time to report it 👍 
I have plan to remove `process` completely in the new major version and switch to having `development` and `production` build inside the NPM package similar to how it's done by React.

**Comment by schrotie** [comment]
Temporary workaround until this fix is in the release:

Import an ES module with this content _before_ importing GraphQL:
`window.process = {env: {}};`

**Comment by rowthan** [comment]
> Temporary workaround until this fix is in the release:
> 
> Import an ES module with this content _before_ importing GraphQL:
> `window.process = {env: {}};`

it seems not work?
```
window.process = {env: {}};
process = {env:{}}
import { graphql, buildSchema } from 'graphql';
```

**Comment by yaacovCR** [comment]
Closed by #3501
